docs/typescript.md

```diff
@@ -119,7 +133,8 @@ const Image1 = styled('div')<ImageProps>({
 }));
 ```
 
-* The generic type version only works with object styles due to https://github.com/Microsoft/TypeScript/issues/11947.
+* The generic function version only works with object styles in TS <= 2.8 due to https://github.com/Microsoft/TypeScript/issues/11947.
+* If you use TS > 2.9, generic function will works with string styles too, but it will break VSCode syntax highlighting. See https://github.com/emotion-js/emotion/issues/721#issuecomment-396954993
```

Comment:
Could you replace `generic function will works with string styles too` with `generic functions will work with string styles too`?

Reply:
@mitchellhamilton You mean `generic function will work with string styles too`, right? I will fix them.

Reply:
Oh, my mistake! I will fix it